tests/test_api.py

```diff
@@ -1575,7 +1643,7 @@ def test_scratch_build_nodeselector(self, nodeselector_str, nodeselector_dict):
         updated_build_json['metadata']['labels']['scratch'] = 'true'
         updated_build_json['spec']['serviceAccount'] = 'builder'
         if nodeselector_dict:
-            updated_build_json['spec']['nodeSelector'] = nodeselector_dict
+            updated_build_json['spec']['nodeSelector'].update(nodeselector_dict)
```

Comment:
Thanks! I knew these tests were dubious. okay, asserts added, stuff changed to make the asserts pass.